storage_service/common/tests/test_middleware.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import absolute_import
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.test.client import Client
+
+
+class AuditLogMiddlewareTestCase(TestCase):
+    def setUp(self):
+        self.client = Client()
+        User = get_user_model()
+        self.user = User.objects.create_user(username="testclient", password="test")
+        self.client.force_login(self.user)
+
+    def test_audit_log_middleware_adds_username(self):
+        with self.modify_settings(
+            MIDDLEWARE={"append": "common.middleware.AuditLogMiddleware"}
+        ):
+            response = self.client.get("/")
+            self.assertTrue(response.has_header("X-Username"))
+            self.assertEqual(response["X-Username"], self.user.username)
+
+    def test_audit_log_middleware_unauthenticated(self):
+        with self.modify_settings(
+            MIDDLEWARE={"append": "common.middleware.AuditLogMiddleware"}
+        ):
+            self.client.logout()
+            response = self.client.get(settings.LOGIN_URL)
+            self.assertTrue(response.has_header("X-Username"))
+            self.assertEqual(response["X-Username"], "None")
```

Comment:
Is the person logging out really `None`? Some test `docstrings` here for each of the functions would be nice in terms of understanding the test's intentions if that's possible? I feel like my question is a stupid one, but it doesn't seem unreasonable that it could come up again. Someone was doing the logging out after all. I am sure it is covered in the project use-cases.

Reply:
@ross-spencer Good question, and not unreasonable at all! You're right that the person logging out is not `None`, and that the username would be recorded in that case.

The `self.client.logout()` on line 28 here is basically test setup - ensuring that there is not an authenticated user for the next request on line 29, `response = self.client.get(settings.LOGIN_URL)`, which is the one we're actually testing. I'll see if I can re-structure the test and/or clarify in the docstring as you suggest.

Reply:
> I'll see if I can re-structure the test and/or clarify in the docstring as you suggest.

A docstring should suffice.